hudi-common/src/main/java/org/apache/hudi/common/fs/FSUtils.java

```diff
@@ -202,7 +202,13 @@ public static String getFileId(String fullFileName) {
   public static String getRelativePartitionPath(Path basePath, Path fullPartitionPath) {
     basePath = Path.getPathWithoutSchemeAndAuthority(basePath);
     fullPartitionPath = Path.getPathWithoutSchemeAndAuthority(fullPartitionPath);
+
     String fullPartitionPathStr = fullPartitionPath.toString();
+
+    if (!fullPartitionPathStr.startsWith(basePath.toString())) {
+      throw new IllegalArgumentException("Partition path does not belong to base-path");
+    }
```

Comment:
Add a new test case for this logic?

Reply:
Once you add the new test I'll take another look.

Reply:
Done